eclipse-scout-core/src/layout/logicalgrid/LogicalGridLayoutInfo.ts

```diff
@@ -228,77 +230,67 @@ export class LogicalGridLayoutInfo implements LogicalGridLayoutInfoModel {
           prefw = this.logicalWidthInPixel(cons);
         }
         prefw = Math.floor(prefw);
-        for (j = cons.gridx; j < cons.gridx + cons.gridw && j < this.cols; j++) {
+        for (let j = cons.gridx; j < cons.gridx + cons.gridw && j < this.cols; j++) {
           prefWidths[j] = Math.max(prefWidths[j], prefw);
+          maxWidths[j] = Math.min(maxWidths[j], cons.maxWidth);
           if (cons.weightx === 0) {
             fixedWidths[j] = true;
           }
         }
       }
     }
-    for (i = 0; i < compCount; i++) {
-      cons = this.gridDatas[i];
+    const lc = LayoutConstants;
+    for (let i = 0; i < compCount; i++) {
+      let cons = this.gridDatas[i];
       if (cons.gridw > 1) {
         let hSpan = cons.gridw;
-        let spanWidth = 0;
-        let distWidth;
-        // pref
-        for (j = cons.gridx; j < cons.gridx + cons.gridw && j < this.cols; j++) {
+        let spanWidth = [0, 0, 0];
+        let distWidth = [0, 0, 0];
+        for (let j = cons.gridx; j < cons.gridx + cons.gridw && j < this.cols; j++) {
           if (!fixedWidths[j]) {
-            spanWidth += prefWidths[j];
+            spanWidth[lc.PREF] += prefWidths[j];
+            spanWidth[lc.MAX] += maxWidths[j];
           }
         }
+        let hGaps = (hSpan - 1) * this.hgap;
         if (cons.widthHint > 0) {
-          distWidth = cons.widthHint - spanWidth - (hSpan - 1) * this.hgap;
+          distWidth[lc.PREF] = cons.widthHint - spanWidth[lc.PREF] - hGaps;
         } else if (cons.useUiWidth) {
-          distWidth = compSize[i].width - spanWidth - (hSpan - 1) * this.hgap;
+          distWidth[lc.PREF] = compSize[i].width - spanWidth[lc.PREF] - hGaps;
         } else {
-          distWidth = this.logicalWidthInPixel(cons) - spanWidth - (hSpan - 1) * this.hgap;
+          distWidth[lc.PREF] = this.logicalWidthInPixel(cons) - spanWidth[lc.PREF] - hGaps;
         }
-        if (distWidth > 0) {
-          let equalWidth = Math.floor((distWidth + spanWidth) / hSpan);
-          let remainder = (distWidth + spanWidth) % hSpan;
-          let last = -1;
-          for (j = cons.gridx; j < cons.gridx + cons.gridw && j < this.cols; j++) {
-            last = j;
-            if (!fixedWidths[j]) {
-              prefWidths[j] = Math.max(equalWidth, prefWidths[j]);
-            }
-            if (cons.weightx === 0) {
-              fixedWidths[j] = true;
-            }
-          }
-          if (last > -1) {
-            prefWidths[last] += remainder;
-          }
+        if (distWidth[lc.PREF] > 0) {
```

Comment:
After simplifying the above, this needs to be: 
```suggestion
        if (distWidth[lc.PREF] > spanWidth[lc.PREF]) {
```